basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
             return constrained_points;
         }
 
+        std::vector<lanelet::BasicPoint2d> create_lanechange_route(int ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state)
+        {
+            std::vector<lanelet::BasicPoint2d> centerline_points = {};
+
+            //Get starting lanelet and ending lanelets
+            auto lanelets = wm->getLaneletsBetween(starting_downtrack, ending_downtrack,true);
+            lanelet::ConstLanelet starting_lanelet =lanelets.front();
```

Comment:
Worth either checking for, or documenting an assumption or pre-condition, but this is an unchecked access into a vector that's not 100% guaranteed to have elements in it. If the above call fails for any reason (bad map, invalid maneuver) this call will segfault for accessing lanelets[0].

Reply:
Any plans to address this?

Reply:
This was updated so we query the lanelet directly from the world model based on the lanelet id